src/Database/Type/DecimalType.php

```diff
@@ -158,7 +158,12 @@ public function marshal($value)
             return (float)$value;
         }
 
-        return null;
+        // allow custom decimal format (@see https://github.com/cakephp/cakephp/issues/12800)
+        if (preg_match('/[^0-9,. ]/', $value)) {
```

Comment:
Aren't there some locales which use single/double quotes too as thousands separator?

Reply:
For degrees lat/lng maybe :) But for decimal I think this is fair enough here.